The Asus Eee PC prints this during bootup..

ata2.00: limited to UDMA/33 due to 40-wire cable
Which is unfortunate, as it has..
ata2.00: ATA-4: SILICONMOTION SM223AC, , max UDMA/66

I believe there actually is no cable, and the flash is directly
connected to the controller.

[root@eee ~]# hdparm -t /dev/sda

/dev/sda:
 Timing buffered disk reads:   64 MB in  3.08 seconds =  20.76 MB/sec

With the patch below it returns..

/dev/sda:
 Timing buffered disk reads:   76 MB in  3.04 seconds =  25.04 MB/sec


Signed-off-by: Dave Jones <davej@redhat.com>

Index: 2008-devel/drivers/ata/ata_piix.c
===================================================================
--- 2008-devel.orig/drivers/ata/ata_piix.c
+++ 2008-devel/drivers/ata/ata_piix.c
@@ -716,6 +716,7 @@ struct ich_laptop {
 
 static const struct ich_laptop ich_laptop[] = {
 	/* devid, subvendor, subdev */
+	{ 0x2653, 0x1043, 0x82d8 },	/* ICH6M on Asus Eee PC */
 	{ 0x27DF, 0x0005, 0x0280 },	/* ICH7 on Acer 5602WLMi */
 	{ 0x27DF, 0x1025, 0x0102 },	/* ICH7 on Acer 5602aWLMi */
 	{ 0x27DF, 0x1025, 0x0110 },	/* ICH7 on Acer 3682WLMi */
